sql/create_indicators_table.sql

```diff
@@ -1,48 +1,117 @@
+			) AS hpd_viol_all__bldg_month,
+
+		    count(*) FILTER (
+				WHERE inspectiondate between email_date_start_1_week and email_date_end
+			) AS hpd_viol_all__week,
+
+		    count(*) FILTER (
+				WHERE inspectiondate between email_date_start_6_month and email_date_end
+					AND class = any('{B,C}')
+			) AS hpd_viol_bc__6mo,
+
+		    count(*) FILTER (
+				WHERE inspectiondate between email_date_start_1_month and email_date_end
+					AND class = any('{B,C}')
+			) AS hpd_viol_bc__1mo,
+
+		    count(*) FILTER (
+		    	WHERE inspectiondate between email_date_start_1_week and email_date_end 
+					AND class = any('{B,C}')
+				AND violationstatus = 'Open'
+	    	) AS hpd_viol_bc__week,
+
 			array_agg(DISTINCT apartment) FILTER (
-				WHERE apartment NOT IN ('BLDG', 'BLDNG','BUILDI', 'NA') AND inspectiondate >= (CURRENT_DATE - interval '6' month)
-			) AS hpd_viol_apts__6mo,
+				WHERE inspectiondate between email_date_start_6_month and email_date_end
+					AND class = any('{B,C}')
+					AND apartment NOT IN ('BLDG', 'BLDNG','BUILDI', 'NA') 
+			) AS hpd_viol_bc_apts__6mo,
+
 			array_agg(DISTINCT apartment) FILTER (
-				WHERE apartment NOT IN ('BLDG', 'BLDNG','BUILDI', 'NA') AND inspectiondate >= (CURRENT_DATE - interval '7' day) AND violationstatus = 'Open'
-			) AS hpd_viol_apts__week,
+				WHERE inspectiondate between email_date_start_1_week and email_date_end 
+					AND class = any('{B,C}')
+					AND apartment NOT IN ('BLDG', 'BLDNG','BUILDI', 'NA') 
+					AND violationstatus = 'Open'
+			) AS hpd_viol_bc_apts__week,
+
 			count (DISTINCT date_part('week', inspectiondate::date)) FILTER (
-				WHERE inspectiondate >= (CURRENT_DATE - interval '1' year)
+				WHERE inspectiondate between email_date_start_1_year and email_date_end
+					AND class = any('{B,C}')
 			) AS hpd_viol_distress
-		FROM hpd_violations
-		WHERE class = any('{B,C}') 
+
+		FROM email_dates, hpd_violations
 		GROUP BY bbl
 	),
 	hpd_comp AS (
 		SELECT 
 			bbl,
-		    count(*) FILTER (WHERE receiveddate >= (CURRENT_DATE - interval '6' month)) AS hpd_comp__6mo,
-		    count(*) FILTER (WHERE receiveddate >= (CURRENT_DATE - interval '1' month)) AS hpd_comp__1mo,
-		    count(*) FILTER (WHERE receiveddate >= (CURRENT_DATE - interval '7' day)) AS hpd_comp__week,
+
+		    count(*) FILTER (
+				WHERE receiveddate between email_date_start_6_month and email_date_end
+			) AS hpd_comp__6mo,
+
+		    count(*) FILTER (
+				WHERE receiveddate between email_date_start_1_month and email_date_end
+			) AS hpd_comp__1mo,
+
 		    count(*) FILTER (
-		    	WHERE receiveddate >= (CURRENT_DATE - interval '6' month) AND type = any('{IMMEDIATE EMERGENCY,HAZARDOUS,EMERGENCY}')
+				WHERE receiveddate between email_date_start_1_week and email_date_end
+			) AS hpd_comp__week,
+
+		    count(*) FILTER (
+				WHERE receiveddate between email_date_bldg_month_start and email_date_bldg_month_end
+			) AS hpd_comp__bldg_month,
```

Comment:
i think it might be good to put in a comment here that 
`hpd_comp__1mo` is a 30 day range from send date and `hpd_comp__bldg_month` is 1st of the month to date and this logic applies to a couple other indicators with the same naming convention

Reply:
oh right yeah great point, I'll add a comment to the top about the naming convention for all these variations